### DISCUSSION TITLE
von OpenDTU auf OpenDTU-onBattery -> [HTTP ERROR] Bad Request

### DISCUSSION THREAD

**Discussion by 3joern** [hoylabs/OpenDTU-OnBattery#786] [Q&A] [answered]
Moin Zusammen,
ich versuche seit ein paar Monaten auf meinem OpenDTU über das Firmware Update OpenDTU-onBattery zu installieren, bekomme aber jedes mal eine Fehlermeldung: [HTTP ERROR] Bad Request.
Habe das Teil über eBay vorinstalliert gekauft, normale OpenDTU Firmwareupdates funktionieren aber problemlos, hat jemand eine Idee, wie ich OpenDTU-onBattery auf das Ding bekommen?

Hello everyone,
I've been trying for a few months to install OpenDTU-onBattery on my OpenDTU via the firmware update, but every time I get an error message: [HTTP ERROR] Bad Request.
I bought the thing pre-installed on eBay, but normal OpenDTU firmware updates work fine. Does anyone have any idea how I can get OpenDTU-onBattery on the thing?

Salute 
björn

![Bildschirmfoto 2024-03-21 um 13 58 42](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/18442198/f0123729-7922-404e-9c43-8caca83936b7)
![Bildschirmfoto 2024-03-22 um 10 51 37](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/18442198/c09ca263-abd6-44bd-90d7-d449a75850a5)
![Bildschirmfoto 2024-03-21 um 14 18 30](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/18442198/03456458-88c8-410b-b6a3-0792555c51e7)

**Comment by schlimmchen** [comment]
Du hast meiner Einschätzung nach ein altes Paritionslayout (Sketch Größe nur 1856kB) und musst die .factory.bin Variante der Firmware flashen, und zwar über die USB-Schnitstelle deiner Hardware. Dabei gehen deine Einstellungen verloren, weil der Flashspeicher neu partitioniert wird, daher vorher die Einstellungen exportieren und nachher wieder importieren.

**Reply by 3joern** [reply]
Ich versuchs mal, danke. Sehe aber das Sketch insgesamt nur 2mb hat, richtig? die factory.bin hat auch 2mb, kann das Probleme machen?

![Bildschirmfoto 2024-03-22 um 11 47 02](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/18442198/5bae1227-fb66-4236-b9b0-11b92775ec1b)

**Comment by 3joern** [comment]
Sorry, ich brauche kurz Unterstützung. Das ist ständig im Sync Mode, da kommen nur ewig Punkte. Habe das Ganze dann am Laptop versucht (da ich mir nicht sicher war, obs in der VM läuft) dort in den Geräteeinstellungen extra nach dem richtigen Com geschaut und eingesetzt. Sollte alles richtig sein, dennoch nur Punkte. Kein Fortschrittsbalken. Was läuft da schief?

**Reply by schlimmchen** [reply]
Sorry, da bin ich raus, da kann ich nichts beitragen. Ich nutze esptool auf Linux.

Du musst bitte die Binary mit dem "factory.bin" im Namen nehmen. Nur die hat das vollstänige Flash-Layout und die neue Partitionstabelle. Die muss ab Adresse 0 geschrieben werden. Aber das scheinst du richtig eingestellt zu haben.

Hast du denn die passende USB-Schnittstelle an die VM weitergeleitet?

**Reply by spcqike** [reply]
Syncmode klingt so, als ob du GPIO 0 (war es 0?) nicht auf low ziehst während du ihn per usb verbindest. 

Muss der esp nicht in den Flash Modus gebracht werden? Fertige dev Kits machen das on Board automatisch. Wenn in deinem Modul lediglich der nackte esp auf eine Platine gelötet ist, musst du den Flash Mode erst erzwingen.

**Reply by schlimmchen** [reply]
Guter Punkt, @3joern hat ja ein uns unbekanntes Board von ebay gekauft. Das kann womöglich nicht per RTS/DTS (oder welches Signal auch immer genutzt wird) den ESP zurücksetzen und den Bootlader aktivieren. Dann muss der BOOT Pin festgehalten werden, und das Board resettet werden (oder USB (Power-Supply)) einstecken.

**Reply by 3joern** [reply, marked as answer]
Jupp, das wars, habs aufgeschraubt und boot beim Installprocess gedrückt, dann lief es. Vom Verkäufer kam vorher auch noch eine Detailierte beschreibung, damit hats dann geklapppt, Yeahh, Danke Euch.
–––––––––––––––
Hier nochmal die Beschreibung, falls jemand ein Ähnliches Problem hat:
Config unter Einstellungen auf Rechner sichern

Bitte mit USB A - C Daten Kabel die DTU mit deinem PC oder Laptop Verbinden
 
Dann mit einem Chrome Browser den Link öffnen.
 
Hier der Link : https://www.opendtu.solar/firmware/webinstall/ hier wählst du EDP32 aus und geht’s auf Installieren
(hier kam erst eine Fehlermeldung, die sich mit klick auf Install und gleichzeitig den Bootbutton erledigt hat)
 
Jetzt diese .bin Datei herunterladen und in Download ablegen [Link entfernt].
 
Dann Verbindest du dich mit deiner DTU wieder per W-Lan und gehst auf EINSTELLUNGEN - FIRMWARE UPDATET und nimmst diese Datei.
 
DATEI AUSWÄHLEN : opendtu-onbattery-generic.bin und Updaten.
 
Nach dem Neustarten hast du eine "OnBattery OpenDTU"
 
Wenn das so ist dann hast du bis jetzt alles richtig gemacht. 👍
 
Dann nimmst du die .json Datei aus dem Anhang gehst in die EINSTELLUNGEN -- Konfigurationsverwaltung  und gehst nach den Bildern vor:
 
Wenn alles geklappt hat, zeigt dein Display wieder was an und du kannst deine OnBattery DTU nutzen.

**Reply by schlimmchen** [reply]
Ich habe den Link zu nordicskill entfernt, weil das meiner Meinung nach keine Quelle sein sollte, um OpenDTU Firmware zu laden. Da kannst sonst was drinstecken. Bitte nimm die auf github als Release veröffentlichen Artefakte.

**Reply by 3joern** [reply]
Danke und hab gleich noch den aktuellsten Build hinterher installiert. Jetzt kämpf ich mit der richtigen URL und Json für mein Tasmota Lesekopf. Aber das kläre ich lieber an anderer Stelle. Danke Euch nochmals.
